polyfill/lib/ecmascript.mjs

```diff
@@ -211,6 +224,54 @@ function getIntlDateTimeFormatEnUsForTimeZone(timeZoneIdentifier) {
 }
 
 export const ES = ObjectAssign({}, ES2020, {
+  // copied from es-abstract/2020/CopyDataProperties.js
+  // with modifications per Temporal spec/mainadditions.html
+  CopyDataProperties: (target, source, excludedKeys, excludedValues) => {
+    if (Type(target) !== 'Object') {
+      throw new $TypeError('Assertion failed: "target" must be an Object');
+    }
+
+    if (!IsArray(excludedKeys) || !every(excludedKeys, IsPropertyKey)) {
+      throw new $TypeError('Assertion failed: "excludedKeys" must be a List of Property Keys');
+    }
+
+    if (typeof source === 'undefined' || source === null) {
+      return target;
+    }
+
+    var from = ToObject(source);
+
+    var sourceKeys = OwnPropertyKeys(from);
+    forEach(sourceKeys, function (nextKey) {
+      var excluded = false;
+
+      forEach(excludedKeys, function (e) {
+        if (SameValue(e, nextKey) === true) {
+          excluded = true;
+        }
+      });
+
+      var enumerable = $isEnumerable(from, nextKey) || (
+      // this is to handle string keys being non-enumerable in older engines
```

Comment:
I'm not overly concerned about handling this case in the online playground... we already require that the browser supports WeakMap, for example. I feel we can leave this support to other polyfills more intended for that use case in production.